libs/labelbox/src/labelbox/client.py

```diff
@@ -214,15 +214,20 @@ def convert_value(value):
             if files:
                 del headers['Content-Type']
                 del headers['Accept']
+            
             request = requests.Request('POST',
                                        endpoint,
                                        headers=headers,
                                        data=data,
                                        files=files if files else None)
 
-            prepped: requests.PreparedRequest = request.prepare()
-
-            response = self._connection.send(prepped, timeout=timeout)
+            prepped: requests.PreparedRequest = self._connection.prepare_request(
```

Comment:
I do not believe we should be using `prepare_request` on the Session itself... unfortunately so. This is because we are modifying headers above and if you use prepare_request on the session it will prb merge those headers back.

Reply:
So just to test I printed out both the request headers and prepped headers
```
{
    "User-Agent": "python-requests/2.32.3",
    "Accept-Encoding": "gzip, deflate",
    "Accept": "application/json",
    "Connection": "keep-alive",
    "Authorization": "Bearer <API_KEY>",
    "Content-Type": "application/json",
    "X-User-Agent": "python-sdk 4.0.0",
    "X-Python-Version": "3.8.18-final",
}
{
    "User-Agent": "python-requests/2.32.3",
    "Accept-Encoding": "gzip, deflate",
    "Accept": "application/json",
    "Connection": "keep-alive",
    "Authorization": "Bearer <API_KEY>",
    "Content-Type": "application/json",
    "X-User-Agent": "python-sdk 4.0.0",
    "X-Python-Version": "3.8.18-final",
    "Content-Length": "733",
}
```

Reply:
Ahh I see what your saying. I modified it to add the correct headers to the prepped response. I believe that is what the docs recommended to allow the merge_environment_settings

Reply:
So I swapped it to add the headers to the prepped response instead of on the request maintaining the correct headers